src/network.rs

```diff
@@ -117,119 +131,83 @@ pub struct SwapMetadata {
     taker_peer_id: PeerId,
 }
 
-impl SwapMetadata {
-    pub fn swap_id(&self) -> SwapId {
-        self.swap_id
-    }
-}
-
-#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
-pub struct SwapId {
-    inner: crate::SwapId,
-    shared: SharedSwapId,
-}
-
-impl SwapId {
-    pub fn new(shared: SharedSwapId) -> Self {
-        Self {
-            inner: crate::SwapId::default(),
-            shared,
-        }
-    }
-}
-
-impl From<SwapId> for crate::SwapId {
-    fn from(from: SwapId) -> Self {
-        from.inner
-    }
+// todo: move this from cnd to comit and import it from comit
+#[derive(Debug)]
+pub struct BtcDaiOrderAddresses {
+    pub bitcoin: bitcoin::Address,
+    pub ethereum: ethereum::Address,
 }
 
-/// A `NetworkBehaviour` that delegates to the `Orderbook` and `Comit` behaviours.
+/// A `NetworkBehaviour` that delegates to the `Orderbook` and `SetupSwap` behaviours.
 #[derive(NetworkBehaviour)]
 #[behaviour(out_event = "Event", poll_method = "poll")]
 #[allow(missing_debug_implementations)]
 pub struct Nectar {
-    comit: Comit,
-    orderbook: Orderbook,
+    orderbook: orderbook::Orderbook,
+    setup_swap: setup_swap::SetupSwap<SwapId>,
     #[behaviour(ignore)]
     events: VecDeque<Event>,
+    /// The Bitcoin network we are currently connected to.
     #[behaviour(ignore)]
-    local_peer_id: PeerId,
+    _configured_bitcoin_network: bitcoin::Network,
+    /// The Ethereum network we are currently connected to.
     #[behaviour(ignore)]
-    takers: HashMap<OrderId, PeerId>,
+    _configured_ethereum_network: ethereum::ChainId,
+    /// The address of the DAI ERC20 token contract on the current Ethereum
+    /// network.
     #[behaviour(ignore)]
-    database: Arc<Database>,
+    _dai_contract_address: ethereum::Address,
```

Comment:
I reckon these could easily be features of the wallets that you already have in here.

If you are actually using them, you probably also want to remove the underscore :sweat_smile: